python/cugraph/cugraph/dask/traversal/bfs.py

```diff
@@ -122,12 +122,28 @@ def bfs(input_graph,
 
     if not isinstance(start, (dask_cudf.DataFrame, dask_cudf.Series)):
         if not isinstance(start, (cudf.DataFrame, cudf.Series)):
-            start = cudf.Series(start)
+            vertex_dtype = input_graph.nodes().dtype
+            start = cudf.Series(start, dtype=vertex_dtype)
         if isinstance(start, (cudf.DataFrame, cudf.Series)):
             # convert into a dask_cudf
             start = dask_cudf.from_cudf(start, input_graph._npartitions)
 
     def check_valid_vertex(G, start):
```

Comment:
It's not obvious why we need to define a function here just to call it once as part of the `if check_start` block below.  Should this code just be moved to the `if check_start` block?